src/apps/appstore/app.js

```diff
@@ -393,8 +513,274 @@ define(function(require) {
+							})
+						});
+						self.showMarketplaceConnector(
+							parent
+						);
+					},
+					function() {
+						formThis.prop({ 'disabled': false });
+					});
+			};
+
+			$('.market-settings #save').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+				saveThis.prop({ disabled: true });
+
+				if (monster.ui.valid(configForm)) {
+					const formData = monster.ui.getFormData('configuration_form');
+					marketAction({ action: 'api_url', api_url: formData.api_url }, 'api_url', saveThis);
+				} else {
+					saveThis.prop({ disabled: false });
+				}
+			});
+
+			$('.market-settings #disable').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+
+				self.confirmDisableDialog(function() {
+					saveThis.prop({ disabled: true });
+					marketAction({ action: 'disable' }, 'enabled', saveThis);
+				});
+			});
+
+			$('.market-settings #disconnect').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+
+				self.confirmDisconnectDialog(function() {
+					saveThis.prop({ disabled: true });
+					marketAction({ action: 'unlink' }, 'unlink', saveThis);
+				});
+			});
+		},
+
+		confirmDisableDialog: function(callbackSuccess) {
+			const self = this;
+			const content = self.i18n.active().marketplace.dangerDisableDialog.content;
+			monster.ui.confirm(
+				content,
+				function() {
+					callbackSuccess && callbackSuccess();
+				}
+			);
+		},
+
+		confirmDisconnectDialog: function(callbackSuccess) {
+			const self = this;
+			const appstoreData = self.getStore('appstoreData');
+			const actionKey = self.i18n.active().marketplace.dangerDisconnectDialog.actionKey;
+			const confirmPopup = monster.ui.confirm(
+				$(self.getTemplate({
+					name: 'dangerConfirmDialog',
+					data: {
+						clusterId: appstoreData.marketplace.cluster_id
+					}
+				})),
+				function() {
+					callbackSuccess && callbackSuccess();
+				},
+				null,
+				{
+					title: self.i18n.active().marketplace.dangerDisconnectDialog.title,
+					confirmButtonText: self.i18n.active().marketplace.dangerDisconnectDialog.actionBtn,
+					htmlContent: true
+				}
+			);
+
+			confirmPopup.find('#confirm_button').prop('disabled', true);
+
+			confirmPopup.find('#action_input').on('keyup', function() {
+				if ($(this).val() === actionKey) {
+					confirmPopup.find('#confirm_button').prop('disabled', false);
+				} else {
+					confirmPopup.find('#confirm_button').prop('disabled', true);
+				}
+			});
+		},
+
+		renderMarketWelcome: function(parent) {
+			const self = this;
+			const template = $(self.getTemplate({
+				name: 'marketWelcome'
+			}));
+
+			monster.ui.insertTemplate(parent.find('.right-container'), template);
+			self.bindMarketWelcomeEvents(parent, template);
+		},
+
+		bindMarketWelcomeEvents: function(parent) {
+			const self = this;
+
+			parent.find('#market_enable_connector_btn').on('click', function(e) {
+				e.preventDefault();
+
+				this.maybeSetApiUrl();
```

Comment:
`this` in this context will refer to the current function it's in (`function(e) {}`), which does not expose any method named `maybeSetApiUrl`, and throw a TypeError.

That's why the `self` hack is used, to keep a reference to the outer function's `this`.

Reply:
fixed